src/pages/Organization/OrganizationPatients.tsx

```diff
@@ -28,10 +28,56 @@ interface Props {
 
 export default function OrganizationPatients({ id, navOrganizationId }: Props) {
   const { t } = useTranslation();
-  const { qParams, Pagination, advancedFilter, resultsPerPage, updateQuery } =
-    useFilters({ limit: 15, cacheBlacklist: ["patient"] });
+
+  const {
+    qParams,
+    Pagination,
+    advancedFilter,
+    resultsPerPage,
+    clearSearch,
+    updateQuery,
+  } = useFilters({ limit: 15, cacheBlacklist: ["name", "phone_number"] });
+
   const [organization, setOrganization] = useState<Organization | null>(null);
 
+  const searchOptions = [
+    {
+      key: "name",
+      type: "text" as const,
+      placeholder: "Search by name",
+      value: qParams.name || "",
+    },
+    {
+      key: "phone_number",
+      type: "phone" as const,
+      placeholder: "Search by phone number",
+      value: qParams.phone_number || "",
+    },
+  ];
+
+  const handleSearch = useCallback((key: string, value: string) => {
+    const searchParams = {
+      name: key === "name" ? value : "",
+      phone_number:
+        key === "phone_number"
+          ? value.length >= 13 || value === ""
+            ? value
+            : undefined
+          : undefined,
+    };
+    updateQuery(searchParams);
+  }, []);
+
+  const handleFieldChange = () => {
+    {
+      updateQuery({
+        name: undefined,
+        phone_number: undefined,
+      });
+      clearSearch.value = true;
```

Comment:
although may work, not a fan of doing this. `clearSearch` is not a ref inside `useFilters` implementation, it's a state variable.

if the `clearSearch` was every used in a `useEffect`'s dependencies, doing this way would not trigger the `useEffect` since the ref of `clearSearch` wouldn't change and only the value of `clearSearch.value` would change. Could cause unintended behaviours that would be hard to debug later on.

let's expose the `setClearSearch` from `useFilters` instead.